libcodechecker/libhandlers/parse.py

```diff
@@ -162,61 +161,53 @@ def arg_match(options):
     parser.set_defaults(func=__handle)
 
 
-def parse(f, context, metadata_dict, suppress_handler, skip_handler, steps,
-          processed_path_hashes):
+def parse(plist_file, metadata_dict, rh, file_report_map):
     """
     Prints the results in the given file to the standard output in a human-
     readable format.
 
     Returns the report statistics collected by the result handler.
     """
 
-    if not f.endswith(".plist"):
-        LOG.debug("Skipping input file '" + f + "' as it is not a plist.")
-        return {}, set()
+    if not plist_file.endswith(".plist"):
+        LOG.debug("Skipping input file '%s' as it is not a plist.", plist_file)
+        return set()
 
-    LOG.debug("Parsing input file '" + f + "'")
-
-    rh = plist_parser.PlistToPlaintextFormatter(suppress_handler,
-                                                skip_handler,
-                                                context.severity_map,
-                                                processed_path_hashes)
-
-    rh.print_steps = steps
+    LOG.debug("Parsing input file '%s'", plist_file)
 
     # Set some variables of the result handler to use the saved file.
 
     analyzed_source_file = "UNKNOWN"
     if 'result_source_files' in metadata_dict and \
-            f in metadata_dict['result_source_files']:
+            plist_file in metadata_dict['result_source_files']:
         analyzed_source_file = \
-            metadata_dict['result_source_files'][f]
+            metadata_dict['result_source_files'][plist_file]
 
-    files, reports = rh.parse(f)
+    files, reports = rh.parse(plist_file)
 
-    plist_mtime = util.get_last_mod_time(f)
+    plist_mtime = util.get_last_mod_time(plist_file)
 
     changed_files = set()
     for source_file in files:
         if plist_mtime is None:
             # Failed to get the modification time for
             # a file mark it as changed.
             changed_files.add(source_file)
-            LOG.warning(source_file +
-                        ' is missing since the last analysis.')
+            LOG.warning('%s is missing since the last analysis.', source_file)
             continue
 
         file_mtime = util.get_last_mod_time(source_file)
         if file_mtime > plist_mtime:
             changed_files.add(source_file)
-            LOG.warning(source_file +
-                        ' did change since the last analysis.')
+            LOG.warning('%s did change since the last analysis.', source_file)
+    if analyzed_source_file not in file_report_map:
+        file_report_map[analyzed_source_file] = []
+
+    if not changed_files:
+        for report in reports:
+            file_report_map[analyzed_source_file].append(report)
```

Comment:
Using the file in the location of the main section might be better for grouping the results.
It is possible that multiple reports end in a header file (maybe the same) in that case we will group them together instead of printing them for each analyzed cpp file where the bug path begins.